### PR TITLE
REST-64 - PHP - Coding Standards Document

### PR DESCRIPTION
**Pull request by sai-manohar**
- Added installation steps for PHPCS and PHPPHAN
- Added a table to explain rules that are used for PHPCS and PHP PHAN configuration
- Added a sample configuration for both PHPCS and PHPPHAN.

### CODE REVIEW

File: coding-standards/php.md [hunk 1]
```diff
@@ -11,5 +11,317 @@ https://www.php-fig.org/psr/psr-12/
 [https://cheatsheetseries.owasp.org/cheatsheets/Laravel_Cheat_Sheet.html](https://cheatsheetseries.owasp.org/cheatsheets/PHP_Configuration_Cheat_Sheet.html)
```
Comment (osm-vishnukyatannawar) [inline coding-standards/php.md:11]: Laravel do we need still? We decided to have it for basic PHP code first.
If they are common for basic PHP also then fine.

Comment (osm-vishnukyatannawar) [reply coding-standards/php.md:11]: Can we have similar title as that of dotnet?

https://github.com/sai-manohar/dev-standards/blob/main/coding-standards/dotnet.md

- C# coding standards
- .NET security best practices

We can decide a common term later on but for now just replace tech infront and keep the same heading

**Review by osm-vishnukyatannawar — APPROVED** [review]
Looks fine to me.

Let Abijeet also check this once.

File: coding-standards/php.md (hunk 1 above)
Comment (saimanoharp) [reply coding-standards/php.md]: Followed the headings similar to dot net. 
Updated security doc to generic PHP instead of laravel.

**Comment by sai-manohar** [comment]
Merging this for now. Will update this if there are any further comments.

File: coding-standards/php.md [hunk 2]
```diff
@@ -1,15 +1,322 @@
...
-
-## Naming Conventions
-https://www.php-fig.org/psr/psr-12/
+## PHP best practides
+We follow the best practices compiled under [PSR12](https://www.php-fig.org/psr/psr-12/)
 
 ## Security Standards
-[https://cheatsheetseries.owasp.org/cheatsheets/Laravel_Cheat_Sheet.html](https://cheatsheetseries.owasp.org/cheatsheets/PHP_Configuration_Cheat_Sheet.html)
-
+We adopted the security practices provided by [OSWAP Cheat Series](https://cheatsheetseries.owasp.org/cheatsheets/PHP_Configuration_Cheat_Sheet.html) while building the applications using PHP.
```
Comment (Abijeet) [inline coding-standards/php.md:8]: > We follow security practices outlined by

File: coding-standards/php.md [hunk 3]
```diff
@@ -1,15 +1,322 @@
...
+# PHP coding standards
+We adopted the standards for programming in PHP from from [slevomat standards](https://github.com/slevomat/coding-standard#alphabetical-list-of-sniffs) and popular [php phan](https://github.com/phan/phan/tree/v5/.phan/plugins#readme) recommendations. 
 
-## Coding Conventions
-https://github.com/slevomat/coding-standard#alphabetical-list-of-sniffs
-
-## Naming Conventions
-https://www.php-fig.org/psr/psr-12/
+## PHP best practides
+We follow the best practices compiled under [PSR12](https://www.php-fig.org/psr/psr-12/)
```
Comment (Abijeet) [inline coding-standards/php.md:5]: I think we also wanted to mention: https://phptherightway.com/

File: coding-standards/php.md [hunk 4]
```diff
@@ -1,15 +1,322 @@
...
+| `plugins`                          | A list of plugin files to execute.                                 | [Link](https://github.com/phan/phan/wiki/Phan-Config-Settings#plugins)                                  |
+| `suppress_issue_types`             | Inhibits some issues which don't matter much.                      | [Link](https://github.com/phan/phan/wiki/Phan-Config-Settings#suppress_issue_types)                     |
+| `backward_compatibility_checks`    | Backwards Compatibility Checking. It consumes a lot of memory, do only if necessary. | [Link](https://github.com/phan/phan/wiki/Phan-Config-Settings#backward_compatibility_checks)       |
+| `unused_variable_detection`        | Set to true in order to attempt to detect unused variables.         | [Link](https://github.com/phan/phan/wiki/Phan-Config-Settings#unused_variable_detection)               |
+
+
+All the phan settings can be found [here](https://github.com/phan/phan/wiki/Phan-Config-Settings)
+
+
+PHP Phan works with plugins which comes along while installing it. Here are some plugins that we use
```
Comment (Abijeet) [inline coding-standards/php.md:180]: These are good additions. Good job!

File: coding-standards/php.md [hunk 5]
```diff
@@ -1,15 +1,322 @@
...
+
+```json
+"scripts": {
+    // ..<existing scripts>
+    "phan": "vendor/bin/phan --config-file .phan/config.php",
+  },
+```
+**`composer run phan`**
+
+This command will analyze and display all the issues if any.
```
Comment (Abijeet) [inline coding-standards/php.md:322]: Do we also want to add what editor and what configurations for that editor that we recommend using?

File: coding-standards/php.md [hunk 6]
```diff
@@ -1,15 +1,322 @@
...
+"scripts": {
+    // ..<existing scripts>
+    "lint": "vendor/bin/phpcs --standard=.phpcs.xml",
+    "lint:fix": "vendor/bin/phpcbf --standard=.phpcs.xml",
+  },
+```
+
+**`composer run lint`**
+
+This command will tokenize PHP, JavaScript and CSS files to detect violations of a defined coding standard.
```
Comment (Abijeet) [inline coding-standards/php.md:151]: Only PHP right? Not JS and CSS.

File: coding-standards/php.md [hunk 7]
```diff
@@ -1,15 +1,322 @@
...
+
+This command will automatically correct coding standard violations.
+
+### PHP PHAN
+#### Installation 
+
+```json
+composer require phan/phan --dev
+```
+Inorder to let your project use php phan, you have to create a folder `.phan` and a file in it `config.php`. All the settings for phan are placed there.
```
Comment (Abijeet) [inline coding-standards/php.md:163]: In some places we are using PHP phan, and in others php phan...lets try to be uniform.

File: coding-standards/php.md [hunk 8]
```diff
@@ -1,15 +1,322 @@
...
+| `exclude_analysis_directory_list`  | Directory list that will be skipped to check violations.           | [Link](https://github.com/phan/phan/wiki/Phan-Config-Settings#exclude_analysis_directory_list)         |
+| `plugins`                          | A list of plugin files to execute.                                 | [Link](https://github.com/phan/phan/wiki/Phan-Config-Settings#plugins)                                  |
+| `suppress_issue_types`             | Inhibits some issues which don't matter much.                      | [Link](https://github.com/phan/phan/wiki/Phan-Config-Settings#suppress_issue_types)                     |
+| `backward_compatibility_checks`    | Backwards Compatibility Checking. It consumes a lot of memory, do only if necessary. | [Link](https://github.com/phan/phan/wiki/Phan-Config-Settings#backward_compatibility_checks)       |
+| `unused_variable_detection`        | Set to true in order to attempt to detect unused variables.         | [Link](https://github.com/phan/phan/wiki/Phan-Config-Settings#unused_variable_detection)               |
+
+
+All the phan settings can be found [here](https://github.com/phan/phan/wiki/Phan-Config-Settings)
+
+
```
Comment (Abijeet) [inline coding-standards/php.md:179]: too much newline here

File: coding-standards/php.md [hunk 9]
```diff
@@ -1,15 +1,322 @@
-# PHP development standards
-This contains PHP standards. 
+# PHP coding standards
+We adopted the standards for programming in PHP from from [slevomat standards](https://github.com/slevomat/coding-standard#alphabetical-list-of-sniffs) and popular [php phan](https://github.com/phan/phan/tree/v5/.phan/plugins#readme) recommendations.
```
[the remark this answers is not in the document]
Comment (saimanoharp) [reply coding-standards/php.md:2]: Updated the same. Addressed this and all the following comments on a different PR #11

**Pull request merged**